evaluations/inkscope-fuzzer_2_dsm-w3f.md

```diff
@@ -0,0 +1,234 @@
+
+The contract was built in DEBUG mode.
+
+Your contract artifacts are ready. You can find them in:
+/home/user/Documents/w3f/inkscope-fuzzer/test-contracts/ityfuzz/target/ink
+
+  - ityfuzz.contract (code + metadata)
+  - ityfuzz.wasm (the contract's code)
+  - ityfuzz.json (the contract's metadata)
+```
+
+I tried to run without changing the version and got this error:
+
+```
+user@localhost:~/Documents/w3f/inkscope-fuzzer/test-contracts/ityfuzz$ docker run -v ".:/contract" inkscope-fuzzer target/ink/ityfuzz.contract
+Error: Failed to parse metadata from the contract file:
+Error("invalid type: string \"4\", expected u64", line: 0, column: 0)
+```
+
+I changed the version, and after that, I could successfully build the contract and run it without error.
+
+```
+user@localhost:~/Documents/w3f/inkscope-fuzzer/test-contracts/ityfuzz$ inkscope-fuzzer target/ink/ityfuzz.contract
+Starting campaign...
+Seed: 0
+Properties found: 4
+Max iterations: 1000
+Fail fast: false
+..❌..................................................................................................
+Campaign status: Optimizing
+Property buggy passed ✅
+Property incr passed ✅
+Property decr passed ✅
+Property inkscope_bug failed ❌
+  Message0: default()
+  Message1: incr(UInt(0))
+  Message2: incr(UInt(0))
+  Message3: incr(UInt(0))
+  Message4: incr(UInt(0))
+  Message5: incr(UInt(2))
+  Message6: incr(UInt(2))
+  Message7: incr(UInt(0))
+  Message8: incr(UInt(0))
+  Message9: decr(UInt(340282366920938463463374607431768211455))
+  Message10: incr(UInt(1))
+  Message11: incr(UInt(1))
+  Message12: incr(UInt(2))
+  Message13: incr(UInt(2))
+  Message14: incr(UInt(1))
+  Message15: incr(UInt(2))
+  Message16: incr(UInt(1))
+  Message17: incr(UInt(2))
+  Message18: buggy()
+  Property: inkscope_bug()
+Campaign finished
+Elapsed time: 36.910255522s
+
+The command bellow show the information via Interface on the terminal.
+```
+
+I was able to see the interface on the terminal using the parameter `-t`
+
+![image](https://github.com/dsm-w3f/Grant-Milestone-Delivery/assets/112647953/c8b3f2d9-52c8-40d6-958c-f5947c328b96)
+
+![image](https://github.com/dsm-w3f/Grant-Milestone-Delivery/assets/112647953/eb9a880b-476b-4264-8530-19756677936a)
+
+
+I tried to run the other examples from Coinfabrik. For these examples, I changed the version too.
+
+For `assert-violation`, I received the error below on both examples.
+
+```
+user@localhost:~/Documents/w3f/inkscope-fuzzer/test-contracts/coinfabrik-test-contracts/assert-violation/remediated-example$ inkscope-fuzzer target/ink/assert_violation.contract
+Starting campaign...
+Seed: 0
+Properties found: 0
+Max iterations: 1000
+Fail fast: false
+thread 'main' panicked at src/engine.rs:1025:66:
+called `Option::unwrap()` on a `None` value
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+```
+
+For `integer-overflow-or-underflow-1`, I received the error below when building the vulnerable example.
```

Comment:
Latest cargo contract versions use clippy to check some rules on the contract. We have added an exception `#[allow(clippy::arithmetic_side_effects)]` in this contracts in order to be able to compile and test this property. 

This is intentional, as we want to test the fuzzer's ability to detect integer overflow or underflow issues in the contract code.